script.js

```diff
@@ -1,12 +1,309 @@
+  query.split(" ").map((character) => {
+    let counter = 0;
+    for (let i = 0; i < showCardEls.length; i++) {
+      if (
+        showNameEls[i].innerText.toLowerCase().indexOf(character) != -1 ||
+        showSummaryEls[i].innerText.toLowerCase().indexOf(character) != -1
+      ) {
+        showCardEls[i].style.display = "";
+        counter++;
+      } else {
+        showCardEls[i].style.display = "none";
+      }
+      spanEl.textContent = `${counter}/${showCardEls.length} showing`;
+    }
+  });
+}
+
+// episodes dropdown feature
+function dropDownResults(e) {
+  const movieTitleEls = document.querySelectorAll(".episode-name");
+  const episodeCardEls = document.querySelectorAll(".episode-card");
+  for (let i = 0; i < movieTitleEls.length; i++) {
+    if (
+      e.target.value == movieTitleEls[i].innerText ||
+      e.target.value == "Main-page"
+    ) {
+      episodeCardEls[i].style.display = "";
+      const spanEl = document.querySelector("span");
+
+      spanEl.textContent = `1/${movieTitleEls.length} episodes showing`;
+    } else {
+      episodeCardEls[i].style.display = "none";
+    }
+  }
+}
+
+//clear the search
+function clearSearch() {
+  const searchElem = document.getElementById("search");
+  searchElem.value = "";
+}
+//episodes drop down event listener
+
+const dropDownElem = document.getElementById("drop-down");
+dropDownElem.addEventListener("change", (e) => {
+  clearSearch();
+  dropDownResults(e);
+});
+//-----------------shows dropdown feature---------------------//
+
+const showsDropDownElem = document.getElementById("shows-dropdown");
+
+showsDropDownElem.addEventListener("change", (e) => {
+  let allShows = sortShows();
+  allShows.forEach((show) => {
+    if (e.target.value === show.name) {
+      getShowsEpisodes(show.id);
+    } else if (e.currentTarget.value === "Shows-Main-page") {
+      setup();
+    }
+  });
+});
+//--------Level 500 shows----------- //
+
+function makePageForShows(showObject) {
+  const rootElem = document.getElementById("root");
+  // const searchElem = document.getElementById("search");
+  // searchElem.value = "";
+  clearSearch();
+  // rootElem.innerHTML = "";
+  clearDisplay();
+
+  showObject.forEach((show) => {
+    const { name, id, rating, image, summary, genres, status, runtime } = show;
+    const showEl = document.createElement("div");
+    showEl.classList.add("show-card");
+    showEl.id = `${id}`; // id of the show used as the id attributes value because it is unique
+    showEl.innerHTML = `
+      <div class="main-info-wrapper">
+        <div class="name-image-wrapper>
+          <div class="name-div">
+            <a href="#" class="name-link" onclick="getShowsEpisodes(${id})"><h1 class="show-name">${name}</h1></a>
+          </div>
+          <img src="${image ? image.medium : ""}" alt="${name}">
+        </div>
+                
+        <div class="show-summary-wrapper" >
+          <p class="show-summary">${summary.split(" ").slice(0, 20).join(" ")}
```

Comment:
not sure why you need to split, split, and join here?

Reply:
I am trying to get the first 'n' elements. It is split->splice->join